pkg/render/common/components/components.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package components
+
+import (
+	operatorv1 "github.com/tigera/operator/api/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type container struct {
+	Name      string                       `json:"name"`
+	Resources *corev1.ResourceRequirements `json:"resources"`
+}
+
+// ApplyDaemonSetOverrides applies the overrides to the given DaemonSet.
+func ApplyDaemonSetOverrides(ds *appsv1.DaemonSet, overrides interface{}) *appsv1.DaemonSet {
```

Comment:
I'm not a huge fan of allowing an `interface{}`as an argument here. I suspect you're doing this because we'll need to handle multiple `XDaemonSet` API structs that have the same fields but different types, but it makes the code a bit messy IMO.

What do you think about an interface-based approach? Something like this maybe: https://go.dev/play/p/QuizDYKfdmr

It would have a few benefits:
- Makes the code easier to follow, and easier to understand what is a valid argument for this function.
- Ensures that each XDaemonSet has a consistent set of fields, requires that they implement the same interface.

Reply:
Yeah I think I was beginning to come around to your suggestion with my [comment](https://github.com/tigera/operator/pull/2063#discussion_r915282729) above

Reply:
What do you think of: a929912d44939dce425bf670df170fbe6145bdcf ?

I wanted the Container type to be an interface too but that leads to an import cycle.

Reply:
Done in a96a441d9cc159e1c9f145ec053523222fdb42a7